packages/web/src/js/Tabs.ts

```diff
@@ -0,0 +1,160 @@
+import BaseComponent from './BaseComponent';
+import EventHandler from './dom/EventHandler';
+import SelectorEngine from './dom/SelectorEngine';
+import { enableToggleTrigger } from './utils/ComponentFunctions';
+import { getElementFromSelector } from './utils/index';
+
+const NAME = 'tabs';
+const DATA_KEY = 'tabs';
+const EVENT_KEY = `.${DATA_KEY}`;
+
+const EVENT_HIDE = `hide${EVENT_KEY}`;
+const EVENT_HIDDEN = `hidden${EVENT_KEY}`;
+const EVENT_SHOW = `show${EVENT_KEY}`;
+const EVENT_SHOWN = `shown${EVENT_KEY}`;
+
+const CLASS_NAME_ACTIVE = 'is-selected';
+
+const SELECTOR_TAB_PANEL = '[role="tablist"]';
+const SELECTOR_OUTER = '[role="presentation"]';
+const SELECTOR_INNER = '[role="tab"]';
+const SELECTOR_DATA_TOGGLE = `[data-toggle="${NAME}"]`;
+const SELECTOR_INNER_ELEM = `${SELECTOR_INNER}, ${SELECTOR_DATA_TOGGLE}`;
+
+class Tabs extends BaseComponent {
+  parent: HTMLElement;
+
+  constructor(element: HTMLElement) {
+    super(element);
+    this.parent = (this.element as Element).closest(SELECTOR_TAB_PANEL) as HTMLElement;
+
+    if (!this.parent) {
+      return;
+    }
+
+    Tabs.setInitialAttributes(this.parent, this.getChildren());
+  }
+
+  static get NAME() {
+    return NAME;
+  }
+
+  show() {
+    const innerElem = this.element as HTMLElement;
+    if (Tabs.isElementActive(innerElem)) {
+      return;
+    }
+
+    const active = this.getActiveElem();
+
+    const hideEvent = active ? EventHandler.trigger(active, EVENT_HIDE, { relatedTarget: innerElem }) : null;
+
+    const showEvent = EventHandler.trigger(innerElem, EVENT_SHOW, { relatedTarget: active }) as Event;
+
+    if (showEvent.defaultPrevented || (hideEvent && hideEvent.defaultPrevented)) {
+      return;
+    }
+
+    this.deactivate(active, innerElem);
+    this.activate(innerElem, active);
+  }
+
+  activate(element: HTMLElement | null, relatedElem?: HTMLElement | null) {
```

Comment:
We agreed with @crishpeen that this is a design problem and should be solved in cooperation with the UX team. So I consider this as not part of this PR. This will be solved in future PR.